tests/test_litapi.py

```diff
@@ -79,3 +95,90 @@ def test_batch_unbatch_stream():
         count += 1
 
     assert count == 4, "Should have 4 responses"
+
+
+def test_decode_request():
+    request = {"input": 4.0}
+    api = ls.examples.SimpleLitAPI()
+    assert api.decode_request(request) == 4.0, "Decode request should return the input 4.0"
+
+
+def test_decode_request_with_openai_spec():
+    api = ls.examples.TestAPI()
+    api._sanitize(max_batch_size=1, spec=ls.OpenAISpec())
+    request = ChatCompletionRequest(messages=[{"role": "system", "content": "Hello"}])
+    decoded_request = api.decode_request(request)
+    assert decoded_request[0]["content"] == "Hello", "Decode request should return the input message"
+
+
+def test_decode_request_with_openai_spec_wrong_request():
+    api = ls.examples.TestAPI()
+    api._sanitize(max_batch_size=1, spec=ls.OpenAISpec())
+    with pytest.raises(AttributeError, match="object has no attribute 'messages'"):
+        api.decode_request({"input": "Hello"})
+
+
+def test_encode_response():
+    response = 4.0
+    api = ls.examples.SimpleLitAPI()
+    assert api.encode_response(response) == {"output": 4.0}, 'Encode response returns encoded output {"output": 4.0}'
```

Comment:
let's extend encoding with more inputs types

Reply:
It is implemented by SimpleLitAPI so for more input types, we would be testing the SimpleLitAPI class then